### PR TITLE
feat: enable support for opting in to technical preview updates

### PR DESCRIPTION
**Pull request by galargh**
###### Description

This PR adds support for a new property `source_ref` in the repository targets. When `source_ref` is set, `Deploy` workflow will check out `protocol/.github` at said ref and copy the workflows from there instead of `github.ref` (the branch in which the workflow is running). 

If `source_ref` is not specified, `Deploy` workflow will act exactly as it did before i.e. it will check out `github.ref`.

###### Why now?

We'd like to let users opt-in to Go 1.19 upgrade before we're able to perform the official release (https://github.com/protocol/.github/issues/362).

But I think this will become useful beyond this immediate need, e.g. for continuous testing of the `next` branch. 

###### Testing
- [x] push to `testing` without setting `source_ref` on `protocol/.github-test-target ` - https://github.com/protocol/.github/runs/7674190338?check_suite_focus=true
```
Run actions/checkout@v2
  with:
    path: template-repo
    ref: refs/heads/testing
```
- [x] push to `testing` with `source_ref` set to `next` on `protocol/.github-test-target ` - https://github.com/protocol/.github/runs/7674236444?check_suite_focus=true
```
Run actions/checkout@v2
  with:
    path: template-repo
    ref: next
```

### CODE REVIEW

**Comment by galargh** [comment]
I just thought of one more thing that we should have for this - triggering. Right now, dispatch doesn't trigger on pushes to `next`. And we cannot simply add `next` to https://github.com/protocol/.github/blob/d36759d161d15a9e3c4e018d3a79227e469b8354/.github/workflows/dispatch.yml#L9 because then `dispatch` would kick off `deploy`s with `refs/heads/next`` for `github.ref`.

Maybe, at least for now, it'd be enough to add `workflow_dispatch` trigger to the `dispatch` workflow. We could guard it so that `dispatch` quits immediately if `github.ref` is different than `refs/heads/master` or `refs/heads/testing`.

With that setup, when someone creates a PR to opt-in their repos to updates from a different branch, then it gets the first update immediately after the merge to `master`. The subsequent upgrades happen either when `master` moves or Unified CI maintainers decide to kick off the `dispatch` workflow.

**Comment by galargh** [comment]
I added the `workflow_dispatch` trigger in https://github.com/protocol/.github/pull/365/commits/5e0b0b9be23401ac6b5ac72bfe8513b8998105fc and here's a run on the `testing` branch that was let through the conditional checking `github.ref` https://github.com/protocol/.github/runs/7675207528?check_suite_focus=true

**Review by laurentsenta — APPROVED** [review]

**Pull request merged**
